api/tests/Api/ContentNodes/Storyboard/CreateStoryboardTest.php

```diff
@@ -43,12 +44,11 @@ public function testCreateStoryboardAcceptsValidJson() {
     }
 
     public function testCreateStoryboardAcceptsEmptyJson() {
-        $this->create($this->getExampleWritePayload(['data' => null]));
+        $response = $this->create($this->getExampleWritePayload(['data' => null]));
 
         $this->assertResponseStatusCodeSame(201);
-        $this->assertJsonContains(['data' => [
-            'sections' => [],
-        ]]);
+
+        $this->assertCount(1, $response->toArray()['data']['sections']); // populated with 1 default section
     }
 
```

Comment:
This test fails now when you don't add the content as default of the field:

    public function testCreateStoryboardAcceptsNotExistingJson() {
        $response = $this->create($this->getExampleWritePayload([], ['data']));

        $this->assertResponseStatusCodeSame(201);

        $this->assertCount(1, $response->toArray()['data']['sections']); // populated with 1 default section
    }

Maybe set the value in the constructor of StoryBoard?

Reply:
Ah good catch. Constructor doesn't work (tried before), because in the constructor I don't know if the `$data` property will be populated later. That's why I put the default value into `setData`. Didn't recognize that `setData` is not called at all though if the property doesn't even exist in the payload.

Was anyway an odd design decision, so moved the default fallback now back into the data persister where it also should be.

Fixed with https://github.com/ecamp/ecamp3/pull/2995/commits/0f70e5861572b7fd29ca50b2ba531636c89030e5

Reply:
We have a similar case in the relation Camp -> Periods, where a camp always must have at least one period. How is this problem solved there?

Reply:
There we enforce that the create payload contains a Period.
testCreateCampValidatesMissingPeriods